plugins/anchor-links/index.js

```diff
@@ -241,7 +247,8 @@ function processAlias(node, startIndex = 0) {
 
     // now, we replace all of the old broken up pieces with a single, combined node containing
     // the full text of the alias
-    node.children.splice(startIndex, endIndex, {
+    const deleteCount = endIndex - startIndex + 1
+    node.children.splice(startIndex, deleteCount, {
```

Comment:
This took me a while to debug.... I think previously, this was kind of only working coincidentally?

Based on my understanding of `splice` and of the logic here, we want to replace all nodes between `startIndex` and `endIndex` with a single `text` node. To do this, it seems we should set `deleteCount` to the difference between those indices, plus one.

I've since noticed that this `processAlias` seems to only work if the are no other separate nodes before the expected pattern. I've added these as a separate test case, for later.